skimage/transform/steerable.py

```diff
@@ -0,0 +1,425 @@
+            The first sublist contains the high pass subband.
+            The last sublist contains the low pass subband.
+            Intermediate sublists contain subbands from different orientations.
+
+        Returns
+        -------
+        image : 2-D array
+            Reconstructed image from subbands of Steerable Pyramid decomposition
+        """
+
+        if (self.height != len(coeff)):
+            raise ValueError("Height of coeff should be %d" % self.height)
+
+        if (len(coeff[0]) != 1):
+            raise ValueError("High pass sublist should be of length one")
+
+        if (len(coeff[-1]) != 1):
+            raise ValueError("Low pass sublist should be of length one")
+
+        for i in range(1, self.height - 1):
+            if (self.n_bands != len(coeff[i])):
+                raise ValueError(
+                    "Size of intermediate sublist should be %d" % self.n_bands)
+
+        r, c = coeff[0][0].shape
+        log_rad, angle = _logradius_angle_grid(r, c)
+
+        Xrcos, Yrcos = _rcos_curve(1, -0.5)
+        Yrcos = np.sqrt(Yrcos)
+        YIrcos = np.sqrt(np.abs(1 - Yrcos * Yrcos))
+
+        lo0mask = _point_op(log_rad, YIrcos, Xrcos)
+        hi0mask = _point_op(log_rad, Yrcos, Xrcos)
+
+        tempdft = self._recon_pyr_level(
+            coeff[1:], log_rad, Xrcos, Yrcos, angle)
+
+        hidft = np.fft.fftshift(np.fft.fft2(coeff[0][0]))
+        outdft = tempdft * lo0mask + hidft * hi0mask
+
+        return np.fft.ifft2(np.fft.ifftshift(outdft)).real
+
+
+def _logradius_angle_grid(m, n):
+    """
+    Helper function that create a grid containing radius from center and angle.
+
+    Parameters
+    ----------
+    m, n : int
+        Size of the desired grid
+
+    Returns
+    -------
+    log_rad : [mxn] array
+        Log of the radius with respect to the center of the matrix
+    angle : [mxn] array
+        The angle of the line originated from center of the matrix
+    """
+
+    ctrm = np.ceil((m + 0.5) / 2).astype(int)
+    ctrn = np.ceil((n + 0.5) / 2).astype(int)
+    xv, yv = np.meshgrid((np.array(range(n)) + 1 - ctrn) / (n / 2),
+                         (np.array(range(m)) + 1 - ctrm) / (m / 2),
+                         sparse=True)
+
+    rad = np.sqrt(xv**2 + yv**2)
+    rad[ctrm - 1, ctrn - 1] = rad[ctrm - 1, ctrn - 2]
+    log_rad = np.log2(rad)
+    angle = np.arctan2(yv, xv)
+
+    return log_rad, angle
+
+
+def _rcos_curve(width, position):
+    """
+    Raised cosine 1D curve.
+
+    Parameters
+    ----------
+    width: float
+        width of the transition
+    position: float
+        position of the cut
+
+    Returns
+    -------
+    X, Y: define a 1D raised cosine curve
+
+    """
+    N = 256
+    X = np.pi * np.array(range(-N - 1, 2)) / 2 / N
+
+    Y = np.cos(X)**2
+    Y[0] = Y[1]
+    Y[N + 2] = Y[N + 1]
+
+    X = position + 2 * width / np.pi * (X + np.pi / 4)
+    return X, Y
+
+
+def _point_op(mask, Y, X):
```

Comment:
I have no idea what this function is doing. Please consider renaming it and providing an example in the docstring?